.gitignore

```diff
@@ -0,0 +1 @@
+build/
```

Comment:
True. This helps everyone who follows the convention in the readme, without making things harder for others. After running the commands from the readme, I expected a clean working tree.

Reply:
fwiw, meson upstream now encourages builddir as the default because it's less ambiguous (e.g. `meson builddir` is less confusing than `meson build` which doesn't actually build anything).

> I expected a clean working tree.

With meson, you do have a clean working tree (especially compared to autotools). Anything built is in the specified build directory which can be wiped at any time. You just happened to specify as subdir of your current directory but the actual worktree is clean. `git clean -xdf` will wipe everything, but that's a different command again.

Reply:
I'll go one step further: meson specifically creates a .gitignore in the builddir/ *for* you, on the rationale that it is trivial for meson to know which path is being used by the person building the project, and it's always correct to ignore the entire directory.

The contents of the file look like this:

```
# This file is autogenerated by Meson. If you change or delete it, it won't be recreated.
*
```

In fact, it will also create a `.hgignore` just in case you are a mercurial user:
```
# This file is autogenerated by Meson. If you change or delete it, it won't be recreated.
syntax: glob
**/*
```

Reply:
Reverted.